packages/web/src/app/doc-pages/brand/illustrations-doc/illustrations-exhibit/illustrations-generator/illustrations-generator.component.ts

```diff
@@ -8,29 +8,34 @@ import { IllustrationName } from '../illustrations-exhibit-data';
   standalone: true,
   template: '',
 })
-export class IllustrationsGeneratorComponent implements AfterViewInit, OnChanges {
+export class IllustrationsGeneratorComponent {
   private illustrationName = input.required<IllustrationName | undefined | null>();
   private color = input<IllustrationColor | null>();
 
-  constructor(private el: ElementRef) {}
-
-  ngAfterViewInit(): void {
-    this.updateIllustration();
-  }
-
-  ngOnChanges(): void {
-    this.updateIllustration();
+  constructor(private el: ElementRef) {
+    effect(() => {
+      this.updateIllustration();
+    });
+    effect(() => {
+      this.updateColor();
+    });
```

Comment:
Does the `effect` know that these methods contain a `signal` and thereby know their dependencies? Or do the signals need to be explicitly called inside the `effect`? 👀

Reply:
It seems like it does yeah! When changing the color, the `updateIllustration` function is not called now, but everything works

Reply:
I separated the color updating from the rest of this functionality so that the illustration DOM elements wouldn't be remade every time the color was changed btw